Jenkinsfile

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env groovy
+
+pipeline {
+  agent {
+    label 'huge && bionic && x86_64 && !xenial && !trusty'
+  }
+
+  options {
+    timestamps()
+    timeout(time: 2, unit: 'HOURS')
+    quietPeriod(5)
+  }
+
+  environment {
+    NPROC = "0"
+  }
+
+  stages {
+    stage('Prepare') {
+      steps {
+        sh '''#!/usr/bin/env bash
+        set -exo pipefail
+        source ./run-make-check.sh
+        FOR_MAKE_CHECK=1 prepare
+        '''
+      }
+    }
+
+    stage('Build') {
+      steps {
+        sh '''#!/usr/bin/env bash
+        set -exo pipefail
+        source ./run-make-check.sh
+        configure
+        build tests
+        '''
+      }
+    }
+
+    stage('Test') {
+      parallel {
+        stage('Unit Tests') {
+          steps {
+            sh '''#!/usr/bin/env bash
+            set -exo pipefail
+            source ./run-make-check.sh
+            cd build
+            export CHECK_MAKEOPTS="$DEFAULT_MAKEOPTS --no-compress-output -T Test"
+            run
+            '''
+          }
```

Comment:
One potential improvement in parallelization is building the unittests in the Unit Test stage, so the API tests can start sooner.